biz.aQute.bndlib/src/aQute/bnd/build/Project.java

```diff
@@ -1887,16 +1888,25 @@ boolean isStale(Set<Project> visited) throws Exception {
 			for (Jar jar : jars) {
 				File file = saveBuild(jar);
 				if (file == null) {
-					getInfo(builder);
 					error("Could not save %s", jar.getName());
-					return null;
+				} else {
+					builtFiles.add(file);
+					if (lastModified < jar.lastModified()) {
+						lastModified = jar.lastModified();
+					}
 				}
-				builtFiles.add(file);
-				if (lastModified < jar.lastModified()) {
-					lastModified = jar.lastModified();
+			}
+
+			if (isOk()) {
+				for (Run export : builder.getExportedRuns()) {
+					lastModified = export(builder, builtFiles, lastModified, export);
```

Comment:
Perhaps this loop and the private export method the loop calls should be moved into ProjectBuilder as a method which is called from this site? This way we keep all the building logic in ProjectBuilder but we call it at the proper time in the Project build process?

Reply:
Well, there is a tension because the ProjectBuilder should not touch files. I am also thinking of using the Exporter services instead of hardcoding the model. Just need to check if there is a difference in how they build. This -export supports -profile, but that is useful for the ExecutableJar exporter as well.